tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,289 @@
+    """Find Service whose spec.selector is a subset of the NetworkPolicy podSelector.matchLabels."""
+    matching_services = []
+    for service in Service.get(dyn_client=admin_client, namespace=hco_namespace.name):
+        selector = service.instance.spec.get("selector", {})
+        # Service selector must be contained in the pod labels set
+        if all(network_policy_match_labels.get(k) == v for k, v in selector.items()):
+            matching_services.append(service)
+
+    assert len(matching_services) == 1, (
+        f"Expected exactly 1 Service whose spec.selector is a subset of NetworkPolicy {network_policy_by_name.name} "
+        f"podSelector in namespace {hco_namespace.name}, but found {len(matching_services)}: "
+        f"{[s.name for s in matching_services]}"
+    )
+
+    return matching_services[0]
+
+
+@pytest.fixture
+def service_ip(service_by_network_policy):
+    """Extract clusterIP for testing network policy enforcement."""
+    service_ip = service_by_network_policy.instance.spec.get("clusterIP")
+    assert service_ip, f"Service {service_by_network_policy.name} has no clusterIP"
+    if ":" in service_ip:
+        pytest.fail(f"This test does not support sending traffic to IPv6: {service_ip}")
+    return service_ip
+
+
+@pytest.fixture
+def service_port(service_by_network_policy):
+    """Extract first port for testing network policy enforcement."""
+    ports = service_by_network_policy.instance.spec.get("ports", [])
+    assert len(ports) > 0, f"Service {service_by_network_policy.name} has no ports defined"
+
+    service_port = ports[0].get("port")
+    assert service_port, f"Service {service_by_network_policy.name} first port has no port number"
+    return service_port
+
+
+@pytest.fixture
+def pods_by_network_policy(admin_client, hco_namespace, network_policy_by_name, network_policy_match_labels):
+    """Get list of running pod names targeted by a NetworkPolicy using its podSelector."""
+    label_pairs = [f"{key}={value}" for key, value in network_policy_match_labels.items()]
+    full_label_selector = ",".join(label_pairs)
+    pods = Pod.get(dyn_client=admin_client, namespace=hco_namespace.name, label_selector=full_label_selector)
+    pod_names = [pod.name for pod in pods if pod.status == Pod.Status.RUNNING]
+    assert len(pod_names) > 0, (
+        f"No running pods found matching NetworkPolicy {network_policy_by_name.name} podSelector "
+        f"in namespace {hco_namespace.name}"
+    )
+    pod_names.sort()
+    return pod_names
+
+
+@pytest.fixture
+def ingress_connectivity_baseline(
+    admin_client, hco_namespace, deployed_client_pod, service_ip, service_port, network_policy_match_labels
+):
+    """
+    Test preparation for ingress tests: Apply allow-all NetworkPolicy, verify traffic works, then remove it.
+    This ensures that without restrictive policies, the connection should succeed.
+    """
+    with AllowAllNetworkPolicy(
```

Comment:
i do not understand the purpose of this fixture - iiuc this tests k8s networkpolicy rules.